tests/test_examples.py

```diff
@@ -1313,3 +1314,239 @@ def test_optimize_with_fourier_planar_coil():
     optimizer = Optimizer("fmintr")
     (c,), _ = optimizer.optimize(c, objective=objective, maxiter=200, ftol=0, xtol=0)
     np.testing.assert_allclose(c.compute("length")[1]["length"], 11, atol=1e-3)
+
+
+@pytest.mark.unit
+def test_coilset_geometry_optimization():
+    """Test optimizing coilset around a fixed axisymmetric equilibrium."""
+    ############ optimize coils with fixed eq ##############
+    R0 = 5  # major radius of plasma
+    a = 1.2  # minor radius of plasma
+    phi0 = np.pi / 12  # initial angle of coil
+    offset = 0.75  # target plasma-coil distance
+
+    # circular tokamak
+    surf = FourierRZToroidalSurface(
+        R_lmn=np.array([R0, a]),
+        Z_lmn=np.array([0, -a]),
+        modes_R=np.array([[0, 0], [1, 0]]),
+        modes_Z=np.array([[0, 0], [-1, 0]]),
+    )
+    eq = Equilibrium(Psi=3, surface=surf, NFP=1, M=2, N=0, sym=True)
+
+    # symmetric coilset with 1 unique coil + 7 virtual coils
+    # initial radius is too large for target plasma-coil distance
+    # initial toroidal angle is too small for target coil-coil distance
+    coil = FourierPlanarCoil(
+        current=1e6,
+        center=[R0, phi0, 0],
+        normal=[0, 1, 0],
+        r_n=[a + 2 * offset],
+        basis="rpz",
+    )
+    coils = CoilSet(coil, NFP=4, sym=True)
+    assert len(coils) == 1
+    assert coils.num_coils == 8
+
+    # optimizing for target coil-plasma distance and maximum coil-coil distance
+    plasma_grid = LinearGrid(M=8, zeta=64)
+    coil_grid = LinearGrid(N=8)
+    objective = ObjectiveFunction(
+        (
+            PlasmaCoilsetMinDistance(
+                eq=eq,
+                coils=coils,
+                target=offset,
+                weight=2,
+                plasma_grid=plasma_grid,
+                coil_grid=coil_grid,
+                eq_fixed=True,
+            ),
+            CoilsetMinDistance(
+                coils,
+                target=2 * np.pi * (R0 - offset) / coils.num_coils,
+                grid=coil_grid,
+            ),
+        )
+    )
+    # TODO: use FixCoilCurrent instead
+    # only 2 free optimization variables are coil center position phi and radius r_n
+    constraints = FixParameters(
+        coils, {"current": True, "center": np.array([0, 2]), "normal": True}
+    )
+    optimizer = Optimizer("scipy-trf")
+    [coils_opt_with_eq], _ = optimizer.optimize(
```

Comment:
So you don't have to pass the equilibrium object if the equilibrium is fixed?

Also, when the compute function inside objectives reads coil_params, does it read the params in the right order?

Reply:
- Yes, if eq is fixed, then it is not among the optimizable things and does not need to be passed in
- Yea it does, the JAX pytree stuff makes sure of that I think

Reply:
other note on this, if one of the objectives takes one object and another takes the other, you still will pass in both (in the case of say one objective being just the eq and the other being just the coils)